src/maestral/cli.py

```diff
@@ -1494,3 +1494,113 @@ def log_level(level_name: str, config_name: str) -> None:
             cli.echo(f"Log level: {level_name}")
+
+
+@main.command(
+    section="Maintenance",
+    help="""
+Compare changes of two revisions of a file.
+
+If the second revision is omitted, it will compare the file to the currently downloaded version.
+The specified revisions will be downloaded if necessary. A pager will be used, if there are more than 50 lines.
+""",
+)
+@click.argument("dropbox_path", type=click.Path())
+@click.option(
+    "-v",
+    "--rev",
+    help="Revisions to compare (mulitple allowed to compare two old revs)",
+    multiple=True,
+    default=[],
+)
+@click.option("--no-color", help="Don't use any colors for the diff", is_flag=True)
+@click.option("--no-pager", help="Don't use a pager for output", is_flag=True)
+@catch_maestral_errors
+@existing_config_option
+# If new_version_hash is omitted, use the current version of the file
+def diff(
+    dropbox_path: str, rev: List[str], no_color: bool, no_pager: bool, config_name: str
+) -> None:
+
+    from datetime import datetime
+    from .daemon import MaestralProxy
+
+    # Reason for rel_dbx_path: os.path.join does not like leading /
+    if not dropbox_path.startswith("/"):
+        dropbox_path = "/" + dropbox_path
+
+    def download_and_compare(m: MaestralProxy, old_rev: str, new_rev: str = None):
+        """
+        Download up to two revisions to a local temporary folder
+        and compare them with a 'diff'. Only text files are supported.
+        If an unknown file type was found, everything that doesn't match
+        'text/*', an error message gets printed.
+        """
+
+        diff = m.get_file_diff(dropbox_path, old_rev, new_rev)
+
+        def color(ind: int, line: str) -> str:
+            """
+            Color diff lines.
+            Inspiration for colors was taken from the
+            well known command 'git diff'.
+            """
+
+            if ind < 2:
+                line = click.style(line, bold=True)
+            elif line.startswith("+"):
+                line = click.style(line, fg="green")
+            elif line.startswith("-"):
+                line = click.style(line, fg="red")
+            # Don't highlight these in the intro
+            elif line.startswith("@@ "):
+                line = click.style(line, fg="cyan")
+            return line
+
+        # Color the lines
+        if not no_color:
+            diff = [color(i, l) for i, l in enumerate(diff)]
+
+        # Enter pager if diff is too long
+        if not no_pager:
+            if len(diff) > 30:
+                click.echo_via_pager("".join(diff))
+                return
+
+        click.echo("".join(diff))
+
+    with MaestralProxy(config_name, fallback=True) as m:
+        if len(rev) == 0:
+            entries = m.list_revisions(dropbox_path)
+            dates = []
+            for entry in entries:
+                cm = cast(str, entry["client_modified"])
+                dt = datetime.strptime(cm, "%Y-%m-%dT%H:%M:%S%z").astimezone()
```

Comment:
This actually does not work in Python 3.6 because "Z" is not recognised as an indicator of UTC. I just had to realise this myself. The fasted way around that is to replace "Z" with "+0000":

```
cm = cm.replace("Z", "+0000")
```